src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoTool.kt

```diff
@@ -44,35 +43,33 @@ object AutoTool : Module() {
     @EventHandler
     private val attackListener =
         Listener(
-            EventHook<PlayerAttackEntityEvent> { event: PlayerAttackEntityEvent? -> equipBestWeapon() }
+            { _: PlayerAttackEntityEvent? -> equipBestWeapon() }
         )
 
-    fun equipBestTool(blockState: BlockState) {
+    fun equipBestTool(blockState: BlockState, blockPos: BlockPos) {
+        val previousSlot = mc.player?.inventory?.selectedSlot
         var bestSlot = -1
-        var max = 0.0
+        var max = 0f
         for (i in 0..8) {
             val stack = mc.player?.inventory?.getStack(i)
             if (stack != null) {
                 if (stack.isEmpty) continue
             }
-            var speed = stack?.getMiningSpeedMultiplier(blockState)
-            var eff: Int
-            if (speed != null) {
-                if (speed > 1) {
-                    speed += (
-                        if (EnchantmentHelper.getLevel(Enchantments.EFFICIENCY, stack).also {
-                                eff = it
-                            } > 0
-                        ) (eff.toDouble().pow(2.0) + 1) else 0.0
-                        ).toFloat()
-                    if (speed > max) {
-                        max = speed.toDouble()
-                        bestSlot = i
-                    }
-                }
+            // this does not send the selected slot to the server
+            // we do this in order to make calcBlockBreakingDelta use the correct slot
+            equip(i, false)
+            // this function is not actually deprecated. mojang just doesn't know what deprecated means
+            val delta = blockState.block.calcBlockBreakingDelta(blockState, mc.player, mc.world, blockPos)
```

Comment:
If it's marked deprecated, I wouldn't be surprised if it disappears in a version to come. However, for now, instead of taking a jab at mojang, let's just suppress the warning:
```suggestion
            @Suppress("DEPRECATION")
            val delta = blockState.block.calcBlockBreakingDelta(blockState, mc.player, mc.world, blockPos)
```